python/src/integry/resources/functions/types.py

```diff
@@ -124,6 +124,94 @@ def get_langchain_tool[
         )
         return tool
 
+    def get_haystack_tool[
+        T
+    ](
+        self,
+        haystack_tool: Callable[..., T],
+        user_id: str,
+        variables: Optional[dict[str, Any]] = None,
+    ) -> T:
+        """
+        Returns a Haystack tool for the function.
+
+        Args:
+            haystack_tool: This should be Haystack Tool method (haystack.tools).
+            user_id: The user ID for authentication.
+
+        Returns:
+            The Haystack tool.
+        """
+
+        schema = self.get_json_schema()
+
+        return haystack_tool(
+            name=schema["name"],
+            description=schema["description"],
+            function=self._get_sync_callable(user_id=user_id, variables=variables),
+            parameters=schema["parameters"],
+        )
+
+    def get_smolagent_tool[
+        T
+    ](
+        self,
+        newTool: Callable[..., T],
+        user_id: str,
+        variables: Optional[dict[str, Any]] = None,
+    ) -> T:
+        def generate_docstring_from_schema(schema: dict[str, Any]) -> str:
+            """
+            Generates a dynamic docstring based on a given JSON schema.
+
+            Parameters:
+            - schema (dict): JSON schema containing parameter definitions.
+
+            Returns:
+            - str: Formatted docstring with parameter details.
```

Comment:
Please ensure that docstring are consistent with rest of the project.

```suggestion
            Args:
                schema: JSON schema containing parameter definitions.

            Returns:
                Formatted docstring with parameter details.
```